teps/0091-trusted-resources.md

```diff
@@ -364,26 +373,110 @@ type KeylessRef struct {
 
 ### Configuration
 
-Reconciler will respond to verification failure based on ConfigMap settings to:
-1) Return error in reconciler, mark the status as error with `FailedVerification` reason;
-2) Not return error, but mark the TaskRun/PipelineRun as `FailedVerification`;
-3) Skip the verification;
-
-Configuration sample:
-```yaml
+ ```yaml
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: feature-flag
   namespace: tekton-pipelines
 data:
-  verification-policy: enforce
+  verification-no-match-policy: "deny"
 ```
 
-`verification-policy`. (Optional, `enforce`, `warn` or `skip`, default `skip`):
- * `enforce`: Failing verification will mark the taskruns/pipelineruns as failed.
- * `warn`: Log warning but don't fail the taskruns/pipelineruns.
- * `skip`: Directly skip the verification.
+`verification-no-match-policy`. (Optional, `allow`, `warn` or `deny`, default to `allow`):
+ * `allow`: Don't fail the taskrun/pipelinerun if no matching policies are found. Don't log.
+ * `warn`: Don't fail the taskrun/pipelinerun and log a warning if no matching policies are found.
+ * `deny`: Fail the taskrun/pipelinerun if no matching policies are found.
+
+ Reconciler will respond to verification failure based on ConfigMap and Verification Policy exsitence settings to:
+1) Return error in reconciler, mark the status as error with `FailedVerification` reason if `enforce` policies fail
+2) Not return error, but warn the log and update the status condition with `FailedVerification` reason if only found `warn` policies.
+3) Skip the verification if no policies exist and `verification-no-match-policy` is set to `allow`.
+
+**Note:** The current proposed `verification-no-match-policy` will replace the old `resource-verification-mode` in one release and this is not a backwards compatible change.
```

Comment:
Could you elaborate a bit more on whether both options will exist for one release or whether `resource-verification-mode` will be removed and `verification-no-match-policy` will be added in the same release? The latter afaik is a backwards incompatible change. The former isn't but it would be good to understand how the options will coexist for that one release

Reply:
We decided with the latter one and I have added the note here. I will update the notice to make it more clear.